### DISCUSSION TITLE
How to make a relation with varchar?

### DISCUSSION THREAD

**Discussion by macleysousa** [prisma/prisma#7361] [Q&A] [answered]
When I'm trying to make a relationship with the `uid` field which is a varchar, the following error appears: 

Error parsing attribute "@relation": The type of the field "user" in the model "bnb_wallet" is not matching the type of the referenced field "uid" in model "user".

 ```prisma
model bnb_wallet {
  user                  Int
  uid                   String   @db.VarChar(20)
  wallet                String   @id @db.VarChar(255)
  private_key           String   @db.VarChar(255)
  created               DateTime @default(now()) @db.DateTime(0)
  user_bnb_walletTouser user     @relation(fields: [user], references: [uid])
}

model user {
  id                      Int          @id @default(autoincrement())
  uid                     String       @unique @db.VarChar(20)
  email                   String       @unique @db.VarChar(255)
  name                    String       @db.VarChar(255)
  type_document           Int?
  identification_document String?      @db.VarChar(255)
  phone_number            String?      @db.VarChar(255)
  level                   Int          @default(1)
  verified                Boolean      @default(false)
  two_factors             Boolean      @default(false)
  bnb_wallet              bnb_wallet[]
}
```

**Comment by janpio** [comment, marked as answer]
Both sides of the relation, so `bnb_wallet.user` and `user.uid` need to be the same type. Right now they are `Int` and `String`.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
